src/action/perf.rs

```diff
@@ -0,0 +1,51 @@
+pub use crate::client::action::perf::WarmConnectionPoolFuture;
+use crate::Client;
+
+impl Client {
+    /// Add connections to the connection pool up to `min_pool_size`.  This is normally not needed -
+    /// the connection pool will be filled in the background, and new connections created as needed
+    /// up to `max_pool_size`.  However, it can sometimes be preferable to pay the (larger) cost of
+    /// creating new connections up-front so that individual operations execute as quickly as
+    /// possible.
+    ///
+    /// Note that topology changes require rebuilding the connection pool, so this method cannot
+    /// guarantee that the pool will always be filled for the lifetime of the `Client`.
+    ///
+    /// Does nothing if `min_pool_size` is unset or zero.
+    ///
+    /// `await` will return `()`.
+    pub fn warm_connection_pool(&self) -> WarmConnectionPool {
+        WarmConnectionPool { client: self }
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl crate::sync::Client {
+    /// Add connections to the connection pool up to `min_pool_size`.  This is normally not needed -
+    /// the connection pool will be filled in the background, and new connections created as needed
+    /// up to `max_pool_size`.  However, it can sometimes be preferable to pay the (larger) cost of
+    /// creating new connections up-front so that individual operations execute as quickly as
+    /// possible.
+    ///
+    /// Note that topology changes require rebuilding the connection pool, so this method cannot
+    /// guarantee that the pool will always be filled for the lifetime of the `Client`.
+    ///
+    /// Does nothing if `min_pool_size` is unset or zero.
+    ///
+    /// `await` will return `()`.
+    ///
+    /// [`run`](WarmConnectionPool::run) will return `()`.
+    pub fn warm_connection_pool(&self) -> WarmConnectionPool {
+        self.async_client.warm_connection_pool()
+    }
+}
+
+/// Add connections to the connection pool up to `min_pool_size`.  Create by calling
+/// [`Client::warm_connection_pool`] and execute with `await` (or [`run`](WarmConnectionPool::run)
+/// if using the sync client).
+#[must_use]
+pub struct WarmConnectionPool<'a> {
+    pub(crate) client: &'a Client,
+}
+
+// IntoFuture impl in src/client/action/perf.rs
```

Comment:
Some of the implementation of the methods used things (e.g. `Client::inner`) that aren't exposed to the rest of the crate.  With the `action` module no longer a submodule of `client`, that didn't work, so the options were:
* make the relevant fields `pub(crate)`
* implement the relevant bits in a submodule and re-export it

While we generally haven't been too careful with crate-interior visibility I went with the second because I'm generally a fan of visibility boundaries as a dev aid.  Not entirely convinced this was the right choice though.